semanticdb/metai/src/main/scala/scala/meta/internal/metai/Main.scala

```diff
@@ -0,0 +1,100 @@
+package scala.meta.internal.metai
+
+import java.net.URI
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.StandardOpenOption
+import scala.collection.mutable
+import scala.meta.cli.Reporter
+import scala.meta.internal.io.FileIO
+import scala.meta.internal.io.PathIO
+import scala.meta.internal.io.PlatformFileIO
+import scala.meta.internal.{semanticidx => i}
+import scala.meta.internal.{semanticdb => s}
+import scala.meta.internal.semanticdb.Scala._
+import scala.meta.internal.io._
+import scala.meta.io.AbsolutePath
+import scala.meta.io.Classpath
+import scala.meta.io.RelativePath
+import scala.meta.metai.Settings
+import scala.util.control.NonFatal
+
+final class Main(settings: Settings, reporter: Reporter) {
+  def process(): Option[Classpath] = {
+    var success = true
+    settings.classpath.foreach { entry =>
+      try {
+        processEntry(entry)
+      } catch {
+        case NonFatal(e) =>
+          e.printStackTrace(reporter.err)
+          success = false
+      }
+    }
+    if (success) {
+      Some(settings.classpath)
+    } else {
+      None
+    }
+  }
+
+  private def processEntry(entry: AbsolutePath): Unit = {
+    val semanticdb = entry.resolve("META-INF").resolve("semanticdb")
+    if (!semanticdb.isDirectory) return
```

Comment:
Let's error out here to avoid situation when we silently do the wrong thing.

Reply:
Agreed.

Reply:
Done, metai now errors in case no semanticdb files are found unless the entry is a manifest jar.